business_requirement/models/business.py

```diff
@@ -0,0 +1,444 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, fields, models, _
+from openerp.exceptions import except_orm
+from openerp import tools
+from openerp import SUPERUSER_ID
+
+
+class BusinessRequirement(models.Model):
+    _inherit = ['mail.thread', 'ir.needaction_mixin']
+    _name = "business.requirement"
+    _description = "Business Requirement"
+    _order = 'id desc'
+
+    @api.model
+    def _get_default_company(self):
+        company_id = self.env.user._get_company()
+        if not company_id:
+            raise except_orm(
+                _('Error!'),
+                _('There is no default company for the current user!'))
+        return self.env['res.company'].browse(company_id)
+
+    sequence = fields.Char(
+        'Sequence',
+        readonly=True,
+        copy=False,
+        index=True,
+    )
+    name = fields.Char(
+        'Name',
+        required=False,
+        readonly=True,
+        copy=False,
```

Comment:
@elicoidal The use of the sequence in the view "view_business_requirement_tree" seems incorrect to me. You want to order requirements always in the context of a specific project. You'd never want to order requirements in a general list.

Now, if I create 3 requirements for 2 projects and I try to make a project-specific order in this view, the order will never be respected as project-specific.

IMHO if you want to make the sequence work, do it as it is used in the sales order. That is, create a view in the project for business requirements, and add the handle to that view. Then the sequence will be generated on a per-project basis. You might have duplicates on a sequence between two projects, but that's allright. (same as same sequences for two separate sales orders).